compound.py

```diff
@@ -137,72 +138,141 @@ def update_bot_order_volumes(
+    )
     logger.info(
         "Calculated BO volume changed from: %s to %s"
         % (base_order_volume, new_base_order_volume)
     )
     if max_safety_orders >= 1:
+        db.execute(
+            f"UPDATE bots SET lastcalcso = {new_safety_order_volume} WHERE botid = {bot_id}"
+        )
         logger.info(
             "Calculated SO volume changed from: %s to %s"
             % (safety_order_volume, new_safety_order_volume)
         )
+    db.commit()
 
-    error, data = api.request(
-        entity="bots",
-        action="update",
-        action_id=str(thebot["id"]),
-        payload={
-            "bot_id": thebot["id"],
-            "name": thebot["name"],
-            "pairs": thebot["pairs"],
-            "base_order_volume": new_base_order_volume,  # new base order volume
-            "safety_order_volume": new_safety_order_volume,  # new safety order volume
-            "take_profit": thebot["take_profit"],
-            "martingale_volume_coefficient": thebot["martingale_volume_coefficient"],
-            "martingale_step_coefficient": thebot["martingale_step_coefficient"],
-            "max_active_deals": thebot["max_active_deals"],
-            "max_safety_orders": thebot["max_safety_orders"],
-            "safety_order_step_percentage": thebot["safety_order_step_percentage"],
-            "take_profit_type": thebot["take_profit_type"],
-            "strategy_list": thebot["strategy_list"],
-            "active_safety_orders_count": thebot["active_safety_orders_count"],
-            "leverage_type": thebot["leverage_type"],
-            "leverage_custom_value": thebot["leverage_custom_value"],
-        },
+    rounddigits = get_round_digits(thebot["pairs"][0])
+    new_bo = round(new_base_order_volume, rounddigits)
+    logger.debug(
+        f"Calculated BO volume is {new_bo}"
     )
-    if data:
-        rounddigits = get_round_digits(thebot["pairs"][0])
-
-        if max_safety_orders >= 1:
-            logger.info(
-                f"Compounded ₿{profit_sum:0.{rounddigits}f} in profit "
-                f"from {deals_count} deal(s) made by '{bot_name}'\n"
-                f"Changed BO from ₿{base_order_volume:0.{rounddigits}f} to "
-                f"₿{new_base_order_volume:0.{rounddigits}f}\n"
-                f"Changed SO from ₿{safety_order_volume:0.{rounddigits}f} to "
-                f"₿{new_safety_order_volume:0.{rounddigits}f}",
-                True,
-            )
-        else:
-            logger.info(
-                f"Compounded ₿{profit_sum:0.{rounddigits}f} in profit "
-                f"from {deals_count} deal(s) made by '{bot_name}'\n"
-                f"Changed BO from ₿{base_order_volume:0.{rounddigits}f} to "
-                f"₿{new_base_order_volume:0.{rounddigits}f}",
-                True,
-            )
+    #### Check to see whether or not the new, rounded value of the BO is sufficient to be stored, if not, only store in the database
+    if new_bo <= base_order_volume:
```

Comment:
Deze controle wordt nu enkel toegepast voor de BO en dat begrijp ik...ik zit nog even te denken of het voor een SO nodig is ook. Zeker als je geen 1:1 verhouding daarin hebt staan? Nu ik er langer over denk vermoed ik dat de BO dan ook niet genoeg zou veranderen, dus dat het zo moet kunnen. Ik onderdruk even de neiging om deze comment weg te halen, dan mag je bevestigen dat het inderdaad zo goed genoeg is.

Reply:
Mijn overweging was inderdaad: Als de afgeronde BO niet hoger is dan de huidige, dan zal dat voor de SO ook niet gelden, tenzij je een rare BO:SO verhouding hebt... Denk ik
Als je je SO gaat veranderen terwijl de BO niet veranderd, wordt de verhouding tussen deze ook anders, wat je volgens mij ook niet wilt.